### PR TITLE
Added classes which will operate on the board displayed in the GUI

### PR DESCRIPTION
**Pull request by mprtcz**
Most of the classes are just stub classes, with the exception of GameBoardBuilder which has some methods that fill the grid with buttons, so mostly focus on that one.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        this.gridPane = boardController.getBoardGridPane();
+    }
+
+    public GameBoard buildGameBoard() {
+        fillTheBoardWithButtonsAndLabels();
+        return null;
+    }
+
+    private void fillTheBoardWithButtonsAndLabels() {
+        for (int i = 0; i < BOARD_SIZE_WITH_LABELS; i++) {
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: Try to change i and j to row and col - could be easier to read.

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 2]
```diff
@@ -18,12 +23,21 @@
...
+    public GridPane playersGridPane;
+    public VBox opponentsVBox;
+    public Label opponentsLabel;
+    public GridPane opponentsGridPane;
     public Label gameStatusLabel;
-    public TextField nameTextField;
-    public VBox centralVBox;
 
-    public SocketClient socketClient;
+    private SocketClient socketClient;
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: :+1:

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.spanish_inquisition.battleship.common;
+
+/**
+ * @author Michal_Partacz
+ */
+public class AdjacentFieldsCalc {
+
+    public static int translateCoordinatesToIndex(int i, int j) {
+        return -1;
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java]: implement, what is i and j - naming

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 4]
```diff
@@ -51,12 +65,15 @@ public void onFeatureButtonClicked() {
...
     @FXML
     public void onNameTextFieldEntered() {
         this.sendTextToSocket(this.nameTextField.getText());
     }
+
+    private void sendTextToSocket(String text) {
+        if(this.socketClient != null)
+            this.socketClient.sendStringToServer(text);
+        this.game = new Game();
+        this.game.buildPlayersBoard(new BoardController(new GameBoard(this.playersGridPane)));
```
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: Pls be gentle ;_; overlooked this one

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        boardController.getGameBoard().addBoardTileToHashMap(boardIndex, tile);
+        fillGridPaneHeightAndWidthForNode(tile);
+    }
+
+    private void addVerticalLabelOnIndex(int j) {
+        Label boardBoardTileLabel = new Label(String.valueOf(j));
+        Platform.runLater(() -> this.gridPane.add(boardBoardTileLabel, 0, j));
+        fillGridPaneHeightAndWidthForNode(boardBoardTileLabel);
+    }
+
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: int i, int j ? maybe better to name it in clear way.

File: src/main/resources/client/MainMenu.fxml [hunk 6]
```diff
@@ -8,21 +8,44 @@ corresponding methods which will be invoked in the java class specified in tag f
...
+                        <Label fx:id="opponentsLabel" text="Label" />
+                        <GridPane fx:id="opponentsGridPane" alignment="CENTER" VBox.vgrow="ALWAYS">
+                        </GridPane>
+                     </children>
+                  </VBox>
+               </children>
+            </HBox>
          </children>
       </VBox>
    </center>
```
Comment (paulstarbi) [inline src/main/resources/client/MainMenu.fxml]: Well done with this FXML :+1:

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java (hunk 5 above)
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: ALready done in a new commit 👌

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java (hunk 3 above)
Comment (mprtcz) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java]: Soon :hourglass:

**Pull request merged**